nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/csv/CSVHeaderSchemaStrategy.java

```diff
@@ -79,4 +81,12 @@ public RecordSchema getSchema(Map<String, String> variables, final InputStream c
     public Set<SchemaField> getSuppliedSchemaFields() {
         return schemaFields;
     }
+
+    private String normalizeNameForAvro(String inputName) {
+        String normalizedName = inputName.replaceAll("[^A-Za-z0-9_]", "_");
```

Comment:
How about this `[^A-Za-z0-9_]+`? So rather than adding N underscores for continuous non-compatible characters, it adds a single `_`.